main.cpp

```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
+        void sharpen(int rotations);
+        void writeSomething(std::string writing);
+        bool isSharp();
+    };
+};
+
+struct Drill
+{
+    bool isPlugged;
+    bool drillbitAttached;
+    int powerInWatts;
+    char drillType = 'H';
+    float rotationsToSecureBit = 0;
+    float chuckTightness = 0.f;
+
+    Drill();
+    ~Drill();
+
+    void drillHole(int holes);
+    void unplug();
+    void attachDrillbit(float drillbitWidth);
+};
+
+struct Elevator
+{
+
+    int highestFloor = 50;
+    int lowestFloor = -2;
+    double totalWeightCarried = 0.0;
+    bool elevatorArrived = false;
+    double maxWeight;
+    int currentFloor;
+    double lastUserWeight = 0;
+    std::string name;
+
+    Elevator(std::string name);
+    ~Elevator();
+
+    void goToFloor(int designatedFloor);
+    void onboardPassenger(double userWeight);
+    void offboardPassenger();
+    bool canTakePassengerWeight(double userWeight);
+    void updateWeightCarried(double userWeight, bool userExit);
+    bool arrived();
+	
+    struct ElevatorUser
+    {
+        double weight;
+        int destinationFloor;
+        int currentFloor;
+        bool isInsideElevator = false;
+        bool canExit = true;
+
+        ElevatorUser();
+        ~ElevatorUser();
+
+        void callElevator();
+        void enterElevator();
+        void setDestination(int newDestinationFloor);
+        void exitElevator();
+        void takeStairsInstead(int newDestinationFloor);
+    };
+};
+
+struct SchoolBag
+{
+    PencilCase schoolPencilCase;
+
+    SchoolBag();
+    ~SchoolBag();
+
+    void packPencilCase();
+    void unpackPencilCase();
+};
+
+struct Building
+{
+    Elevator elevatorA{"A1"};
+    Elevator elevatorB{"B1"};
+    bool isOpen;
+
+    Building();
+    ~Building();
+
+    void closeBuilding();
+    void openBuilding();
+    bool buildingEmpty();
+    void clearBuilding();
+    void callElevatorAAndTakePassenger(int designatedFloor, double passengerWeight);
+};
+
+// ============Pencil Functions ================
+
+PencilCase::Pencil::Pencil() : canColor(false), shade("2HB")
+{	
+    std::cout << "****** Pencil Created ******" << std::endl;
+}
+
+PencilCase::Pencil::~Pencil()
+{
+    std::cout << "****** Pencil Destroyed ******" << std::endl;
+}
+
+void PencilCase::Pencil::sharpen(int rotations)
+{
+    if(!isSharp())
+    {
+        std::cout << "Sharpening..." << std::endl;
+        for(int i = 0; i < rotations; ++i)
+            sharpness += 0.067f;
+
+        if(sharpness > 1.5f && sharpness < 2.5f)
+        {
+            sharpEnough = true;
+            std::cout << "Pencil sharpened" << std::endl;
+        }	
+        else if(sharpness >= 2.5f)
+        {
+            sharpEnough = true;
+            std::cout << "Super sharp!" << std::endl;
+        }
+        else
+        {
+            sharpEnough = false;
+            std::cout << "Needs more sharpening" << std::endl;
+        }	
+    }
+    else
+        std::cout << "Pencil sharp enough" << std::endl;
+}
```

Comment:
If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [ ] revise